custom_components/pioneer_async/pioneer_avr.py

```diff
@@ -860,11 +862,20 @@ async def set_volume_level(self, volume, zone="1"):
             or (zone != "1" and volume > MAX_VOLUME_ZONEX)
         ):
             raise ValueError(f"volume {volume} out of range for zone {zone}")
-        vol_len = 3 if zone == "1" else 2
-        vol_prefix = str(volume).zfill(vol_len)
-        return await self.send_command(
-            "set_volume_level", zone, prefix=vol_prefix, ignore_error=False
-        )
+        if self.volume_steps:
+            current_volume = self.volume.get(zone)
+            steps = abs(round((volume - current_volume) / 2))
+            for x in range(steps):
+                if volume > current_volume:
+                    await self.volume_up()
```

Comment:
should be `await self.volume_up(zone=zone)` to work with zones other than 1, same for volume_down.

Reply:
Understood